app/src/main/res/layout-land/submitted_answer_item.xml

```diff
@@ -25,68 +25,84 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:gravity="@{viewModel.hasConversationView ? Gravity.END : Gravity.START}"
-    android:orientation="horizontal"
-    app:explorationSplitViewMarginApplicable="@{viewModel.hasConversationView &amp;&amp; viewModel.isSplitView}"
-    app:explorationSplitViewMarginBottom="@{@dimen/margin_0}"
-    app:explorationSplitViewMarginEnd="@{@dimen/margin_24}"
-    app:explorationSplitViewMarginStart="@{@dimen/margin_40}"
-    app:explorationSplitViewMarginTop="@{@dimen/margin_24}"
-    app:explorationViewMarginApplicable="@{viewModel.hasConversationView &amp;&amp; !viewModel.isSplitView}"
-    app:explorationViewMarginBottom="@{@dimen/margin_0}"
-    app:explorationViewMarginEnd="@{@dimen/margin_48}"
-    app:explorationViewMarginStart="@{@dimen/margin_80}"
-    app:explorationViewMarginTop="@{@dimen/margin_24}"
-    app:questionSplitViewMarginApplicable="@{!viewModel.hasConversationView &amp;&amp; viewModel.isSplitView}"
-    app:questionSplitViewMarginBottom="@{@dimen/margin_0}"
-    app:questionSplitViewMarginEnd="@{@dimen/margin_32}"
-    app:questionSplitViewMarginStart="@{@dimen/margin_32}"
-    app:questionSplitViewMarginTop="@{@dimen/margin_24}"
-    app:questionViewMarginApplicable="@{!viewModel.hasConversationView &amp;&amp; !viewModel.isSplitView}"
-    app:questionViewMarginBottom="@{@dimen/margin_0}"
-    app:questionViewMarginEnd="@{@dimen/margin_64}"
-    app:questionViewMarginStart="@{@dimen/margin_64}"
-    app:questionViewMarginTop="@{@dimen/margin_24}">
+    android:orientation="horizontal">
 
-    <TextView
-      android:id="@+id/submitted_answer_text_view"
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Why keep the outer linear layout if it just has one child?

Reply:
It is needed to set the gravity as required, like in practice questions we want answer textView to be at left and in exploration questions, we want this to be shifted to left

Reply:
Why can't we use constraints + custom binding adapters for this purpose?

Reply:
@BenHenning I think we have set width and height of constraintLayout to wrap-content, so this actually works, before adding the linearLayout when I tested the app, then submitted_answer was always in left, after adding LinearLayout this got fixed because we were using the earlier.
And actually, we are not setting gravity to children of constarintLayout, we are setting gravity to constarintLayout itself.
Does that make sense?

Reply:
Sorry no, this is not clear. I did misread the XML earlier--the LinearLayout's gravity is being set, not its children. Why can't also set the gravity of the ConstraintLayout? I'm still not understanding why we need a LinearLayout with one children versus just bringing the ConstraintLayout up to the become the root of the layout file. A LinearLayout with one child is effectively a no-op.

Reply:
@BenHenning Okay, I'll try this. Thanks for suggesting a better approach.

Reply:
Done